docs/examples/Asymmetric Calculation Example.ipynb

```diff
@@ -0,0 +1,828 @@
+    "# node\n",
+    "node = initialize_array(\"input\", \"node\", 3)\n",
+    "node[\"id\"] = np.array([1, 2, 6])\n",
+    "node[\"u_rated\"] = [10.5e3, 10.5e3, 10.5e3]\n",
+    "\n",
+    "# line\n",
+    "line = initialize_array(\"input\", \"line\", 3)\n",
+    "line[\"id\"] = [3, 5, 8]\n",
+    "line[\"from_node\"] = [1, 2, 1]\n",
+    "line[\"to_node\"] = [2, 6, 6]\n",
+    "line[\"from_status\"] = [1, 1, 1]\n",
+    "line[\"to_status\"] = [1, 1, 1]\n",
+    "line[\"r1\"] = [0.25, 0.25, 0.25]\n",
+    "line[\"x1\"] = [0.2, 0.2, 0.2]\n",
+    "line[\"c1\"] = [10e-6, 10e-6, 10e-6]\n",
+    "line[\"tan1\"] = [0.0, 0.0, 0.0]\n",
+    "line[\"i_n\"] = [1000, 1000, 1000]\n",
+    "line[\"r0\"] = [0.25, 0.25, 0.25]\n",
+    "line[\"x0\"] = [0.2, 0.2, 0.2]\n",
+    "line[\"c0\"] = [10e-6, 10e-6, 10e-6]\n",
+    "line[\"tan0\"] = [0, 0, 0]\n",
+    "\n",
+    "# sym load\n",
+    "sym_load = initialize_array(\"input\", \"sym_load\", 1)\n",
+    "sym_load[\"id\"] = [4]\n",
+    "sym_load[\"node\"] = [2]\n",
+    "sym_load[\"status\"] = [1]\n",
+    "sym_load[\"type\"] = [LoadGenType.const_power]\n",
+    "sym_load[\"p_specified\"] = [20e6]\n",
+    "sym_load[\"q_specified\"] = [5e6]\n",
+    "\n",
+    "# asym load\n",
+    "asym_load = initialize_array(\"input\", \"asym_load\", 1)\n",
+    "asym_load[\"id\"] = [7]\n",
+    "asym_load[\"node\"] = [6]\n",
+    "asym_load[\"status\"] = [1]\n",
+    "asym_load[\"type\"] = [LoadGenType.const_power]\n",
+    "asym_load[\"p_specified\"] = [[10e6, 20e6 , 0]] # the 3 phases may have different loads\n",
+    "asym_load[\"q_specified\"] = [[0, 8e6, 2e6]] # the 3 phases may have different loads\n",
+    "\n",
+    "# source\n",
+    "source = initialize_array(\"input\", \"source\", 1)\n",
+    "source[\"id\"] = [10]\n",
+    "source[\"node\"] = [1]\n",
+    "source[\"status\"] = [1]\n",
+    "source[\"u_ref\"] = [1.0]\n",
+    "\n",
+    "# all\n",
+    "asym_input_data = {\n",
+    "    \"node\": node,\n",
+    "    \"line\": line,\n",
+    "    \"sym_load\": sym_load,\n",
+    "    \"asym_load\": asym_load,\n",
+    "    \"source\": source\n",
+    "}"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "e3605c3e",
+   "metadata": {},
+   "source": [
+    "### One-time Asymmetric Power Flow Calculation\n",
+    "\n",
+    "You can call the method `calculate_power_flow` to do a one-time calculation based on the current network data in the model. In this case you should not specify the argument `update_data` as it is used for batch calculation.\n",
+    "\n",
+    "**NOTE: For asymmetric calculations, the argument `symmetric` of `calculate_power_flow` and `assert_valid_input_data`should be set to False.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "44c2de63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Validation (optional)\n",
+    "from power_grid_model.validation import assert_valid_input_data, assert_valid_batch_data\n",
```

Comment:
it's probably a good idea to run the formatter on all cells to fix any such cases

```suggestion
    "from power_grid_model.validation import assert_valid_input_data, assert_valid_batch_data\n",
    "\n",
```